browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchFragment.java

```diff
@@ -56,8 +58,52 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             mRequestCode = Integer.MIN_VALUE;
         }
+        requireActivity().getApplication().registerActivityLifecycleCallbacks(lifecycleCallbacks);
+
     }
 
+    @Override
+    public void onDestroy() {
+        requireActivity().getApplication().unregisterActivityLifecycleCallbacks(lifecycleCallbacks);
+        super.onDestroy();
+    }
+
+    private Application.ActivityLifecycleCallbacks lifecycleCallbacks = new Application.ActivityLifecycleCallbacks() {
+        @Override
+        public void onActivityCreated(Activity activity, Bundle savedInstanceState) { }
+
+        @Override
+        public void onActivityStarted(Activity activity) { }
+
+        @Override
+        public void onActivityResumed(Activity activity) { }
+
+        @Override
+        public void onActivityPaused(Activity activity) { }
+
+        @Override
+        public void onActivityStopped(Activity activity) { }
+
+        @Override
+        public void onActivitySaveInstanceState(Activity activity, Bundle outState) { }
+
+        @Override
+        public void onActivityDestroyed(Activity activity) {
+            // https://stackoverflow.com/questions/39726547/i-want-to-close-chrome-custom-tab-when-action-button-is-clicked
+            // Relaunch ourselves, bringing ourselves to the front, when the BrowserSwitchActivity
+            // self-finishes.
+            if (activity instanceof BrowserSwitchActivity) {
+                Activity myActivity = getActivity();
+                if (myActivity != null) {
+                    Intent initialIntent = myActivity.getIntent();
+                    Intent relaunchIntent = new Intent(initialIntent)
+                            .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                    myActivity.startActivity(relaunchIntent);
```

Comment:
Note, this requires the activity to have `android:launchMode="singleTask"` in `AndroidManifest.xml`

Reply:
@calvarez-ov This sounds like a breaking change, can you clarify what would happen if `android:launchMode="singleTask"` is not included?

Reply:
Quick answer: without "singleTask", the custom tabs are closed, and our activity is brought to the front. But our activity doesn't detect anything about the successful payment to proceed the flow. I'll update here in a bit with more technical details/explanation. I have to investigate :)

Reply:
Without "singleTask": Our activity's `onCreate()` is called, which calls `braintreeFragment = BraintreeFragment.newInstance(this, token)`. This is a new `BraintreeFragment` instance`.

With "singleTask": our activity's `onCreate()` isn't called. Looks like it's the same instance of the activity, so the same instance of the `BraintreeFragment`

Reply:
Without "singleTask", our activity's `onCreate()` is called with `null` for the saved `Bundle`.  So state can't be restored. Also verified that `onNewIntent()`, and `onRestoreInstanceState()` are not called.